grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/fragment/IdFragment.java

```diff
@@ -75,4 +93,9 @@ public double fragmentCost() {
     public boolean hasFixedFragmentCost() {
         return true;
     }
+
+    @Override
+    public boolean operatesOnEdge() {
+        return id.getValue().startsWith(Schema.PREFIX_EDGE);
```

Comment:
In this case, we're doing the right thing. `operatesOnEdge` just indicates "this fragment may operate on edges", not _only_ edges. Tell me if you can think of a way to make that clearer though!